homeassistant/components/switchbee/switch.py

```diff
@@ -0,0 +1,127 @@
+"""Support for SwitchBee switch."""
+import logging
+from typing import Any
+
+from switchbee.api import SwitchBeeDeviceOfflineError, SwitchBeeError
+from switchbee.device import ApiStateCommand, DeviceType
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import SwitchBeeCoordinator
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbee switch."""
+    coordinator: SwitchBeeCoordinator = hass.data[DOMAIN][entry.entry_id]
+    device_types = (
+        [DeviceType.TimedPowerSwitch]
+        if coordinator.switch_as_light
+        else [
+            DeviceType.TimedPowerSwitch,
+            DeviceType.GroupSwitch,
+            DeviceType.Switch,
+            DeviceType.TimedSwitch,
+            DeviceType.TwoWay,
+        ]
+    )
+
+    async_add_entities(
+        Device(hass, device, coordinator)
+        for device in coordinator.data.values()
+        if device.type in device_types
+    )
+
+
+class Device(CoordinatorEntity, SwitchEntity):
+    """Representation of an Switchbee switch."""
+
+    def __init__(self, hass, device, coordinator):
+        """Initialize the Switchbee switch."""
+        super().__init__(coordinator)
+        self._session = aiohttp_client.async_get_clientsession(hass)
+        self._attr_name = f"{device.zone} {device.name}"
+        self._device_id = device.id
+        self._attr_unique_id = f"{coordinator.mac_formated}-{device.id}"
+        self._attr_is_on = False
+        self._attr_available = True
+        self._attr_has_entity_name = True
+
+    @property
+    def available(self) -> bool:
+        """Available."""
+        return self._attr_available
+
```

Comment:
```suggestion
```
This is not needed since you set `self._attr_available` in `_handle_coordinator_update`

Reply:
without it the making the device unavailable didn't work